content/types/_channel_event.textile

```diff
@@ -1,25 +1,92 @@
-@ARTChannelEvent@ is the enum emitted as the event in @ARTRealtimeChannel.on@. The variants correspond to "@Realtime Channel@ state":/realtime/channels-messages/#channel-state changes, plus an arbitrary @Error@.
-
-```[objc]
-  typedef NS_ENUM(NSUInteger, ARTChannelEvent) {
-      ARTChannelEventInitialized,
-      ARTChannelEventAttaching,
-      ARTChannelEventAttached,
-      ARTChannelEventDetaching,
-      ARTChannelEventDetached,
-      ARTChannelEventFailed,
-      ARTChannelEventError
-  };
-```
-
-```[swift]
-  enum ARTChannelEvent : UInt {
-      case Initialized
-      case Attaching
-      case Attached
-      case Detaching
-      case Detached
-      case Failed
-      case Error
-  }
-```
+blang[jsall].
+  @ChannelEvent@ is a String that can be emitted as an event on the @Channel@ object; either a "@ChannelState@":/realtime/channels-messages#channel-states or an @update@ event.
+
+
+  ```[javascript]
+    var allChannelEvents = allChannelStates.concat(['update']);
```

Comment:
I don't think this makes it any easier for a developer i.e. expecting them to jump around documentation to find out what `allChannelStates` is only makes their job harder.  We have had feedback from developers quite often that they feel they need to jump around too much in docs and they'd prefer we duplicate content to help them avoid that.  So my vote is that a partial should not make assumptions about other partials existing close enough to this content for it to make sense and we should go with simplicity for the reader.